infra/ansible/setup-worker.yml

```diff
@@ -0,0 +1,53 @@
+---
+- hosts: workers:&local
+  become: yes
+  gather_facts: yes
+  tasks:
+    - name: Get hosts
+      debug: var=ansible_play_hosts
+    - name: Get hostname
+      debug: var=ansible_facts.hostname
+    - name: Add keys to authorized_keys
+      authorized_key:
+        user: ubuntu
+        state: present
+        key: http://172.16.0.10/authorized_keys
```

Comment:
I guess you are fetching the `authorized_keys` from the `master`, since the prv/pub SSH key for the `master` is currently dynamic and auto-generated during provisioning time.

I am not sure why we need the workers to have SSH access using the master's SSH key - I think we should be just fine by applying a static `authorized_keys` file which contains 3-4 public keys for all team members.

Reply:
+1, this is what we do on the gateways'n'bootstrappers

Reply:
I generally use ssh-agent forwarding, so I can usually run the Ansible scripts on the manager node and it can access the other nodes. Sometimes I forget to add the ssh key to my agent and I see a failure. Where this will be really useful is when we want to run Ansible scripts from Terraform, which won't have ssh-agent.